src/DIRAC/DataManagementSystem/scripts/dirac_dms_filecatalog_cli.py

```diff
@@ -20,50 +20,66 @@
 
-    from DIRAC import gConfig, exit as dexit
-    from DIRAC.Resources.Catalog.FileCatalogFactory import FileCatalogFactory
-
-    fcType = gConfig.getValue("/LocalSite/FileCatalog", "")
-
-    res = gConfig.getSections("/Resources/FileCatalogs", listOrdered=True)
-    if not res["OK"]:
-        dexit(1)
-    fcList = res["Value"]
-    if not fcType:
-        if res["OK"]:
-            fcType = res["Value"][0]
+    from DIRAC import exit as dexit
 
     for switch in Script.getUnprocessedSwitches():
         if switch[0].lower() == "f" or switch[0].lower() == "file-catalog":
             fcType = switch[1]
 
     if not fcType:
-        print("No file catalog given and defaults could not be obtained")
-        sys.exit(1)
+        # A particular catalog is not specified, try to instantiate the catalog container
+        from DIRAC.Resources.Catalog.FileCatalog import FileCatalog
+
+        catalog = FileCatalog()
+        if not catalog.valid:
+            print("Failed to create the FileCatalog container. Try to use a specific catalog with -f option")
+            dexit(-1)
+        result = catalog.getMasterCatalogNames()
+        if not result["OK"]:
+            print("Failed to get the Master catalog name for the FileCatalog container")
+            dexit(-1)
+        masterCatalog = result["Value"][0]
+        readCatalogs = [c[0] for c in catalog.getReadCatalogs()]
+        writeCatalogs = [c[0] for c in catalog.getWriteCatalogs()]
+        allCatalogs = list(set([masterCatalog] + readCatalogs + writeCatalogs))
+
+        if len(allCatalogs) == 1:
+            # If we have a single catalog in the container, let's use this catalog directly
+            fcType = allCatalogs[0]
+            catalog = None
+        else:
+            print("Starting FileCatalog container client:")
+            print(f"   {masterCatalog} - Master")
+            for cat in allCatalogs:
+                if cat != masterCatalog:
+                    cTypes = ["Write"] if cat in writeCatalogs else []
+                    cTypes.extend(["Read"] if cat in readCatalogs else [])
+                    print(f"   {cat} - {'-'.join(cTypes)}")
+            print("")
+
+    if fcType:
+        from DIRAC.Resources.Catalog.FileCatalogFactory import FileCatalogFactory
+
+        result = FileCatalogFactory().createCatalog(fcType)
+        if not result["OK"]:
+            print(result["Message"])
+            dexit(-1)
+        catalog = result["Value"]
+        print(f"Starting {fcType} client")
 
     from DIRAC.DataManagementSystem.Client.FileCatalogClientCLI import FileCatalogClientCLI
 
-    result = FileCatalogFactory().createCatalog(fcType)
-    if not result["OK"]:
-        print(result["Message"])
-        if fcList:
-            print("Possible choices are:")
-            for fc in fcList:
-                print(" " * 5, fc)
-        sys.exit(1)
-    print(f"Starting {fcType} client")
-    catalog = result["Value"]
     cli = FileCatalogClientCLI(catalog)
```

Comment:
so if `catalog` is still None, it does what?

Reply:
catalog can not be None by this point. It is either a specific FileCatalog, e.g. AugerFileCatalog, or a FileCatalog() object which is containing several catalogs as defined for a VO, e.g. DIRAC File Catalog and Bookkeeping in LHCb. In the ILC I think you have also Transformation File Catalog. If the File Catalog resolution somehow failed, the command should have exited already before this point.

Reply:
Well, you added an `if catalog:` now! :D

Reply:
I added it just in case that I am not aware of :)